Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM gcc:9.2
+
+ENV DEBIAN_FRONTEND noninteractive
+
+# Installing Dependencies
+RUN apt-get update && apt-get install -y \
+    wget \
+ && rm -rf /var/lib/apt/lists/* 
+
+# Installing CMake
+RUN wget "https://github.com/Kitware/CMake/releases/download/v3.18.3/cmake-3.18.3-Linux-x86_64.sh" \
+ && chmod a+x cmake-3.18.3-Linux-x86_64.sh \
+ && ./cmake-3.18.3-Linux-x86_64.sh --prefix=/usr/local --skip-license \
+ && rm cmake-3.18.3-Linux-x86_64.sh
+
+# Copying files
+COPY . /app
+WORKDIR /app
```

Comment:
We should probably explicitly specify which directories to copy to avoid copying unwanted files or previous build artefacts to the container.

Reply:
Implemented `.dockerignore` file.

Reply:
Thanks for the updates! Tbh I still prefer explicitly copying the necessary directories and files as opposed to blacklisting.

- The `.dockerignore` does not contain any `build` directories or other non-relevant files that users might create before running docker manually
- Adding the `.dockerignore` creates yet another root level file to the starter and must be maintained
- Explicitly listing files ensures that only relevant files get copied and the CI will complain if we forgot something

Alternatively, we could [copy only files tracked by git](https://stackoverflow.com/questions/51693245/copy-only-git-tracked-files-in-dockerfile) for a clean container state.